tests/basic/storage_pool_test.go

```diff
@@ -5345,20 +5354,33 @@ var _ = Describe("{PoolResizeVolumesResync}", func() {
 					}
 				}(eachVol)
 			}
-			wg.Wait()
-			dash.VerifyFatal(len(error_array) == 0, true, fmt.Sprintf("errored while setting replication on volumes [%v]", error_array))
 
 			log.InfoD("Waiting till Volume is In Resync Mode ")
 			if WaitTillVolumeInResync(randomVolIDs) == false {
 				log.InfoD("Failed to get Volume in Resync state [%s]", randomVolIDs)
 			}
 
 			log.InfoD("Current Size of the pool %s is %d", rebootPoolID, poolToBeResized.TotalSize/units.GiB)
-			err = Inst().V.ExpandPool(rebootPoolID, api.SdkStoragePool_RESIZE_TYPE_AUTO, expectedSize, true)
+			poolExpandLock.Lock()
+			err = Inst().V.ExpandPool(rebootPoolID, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, true)
 			dash.VerifyFatal(err, nil, "Pool expansion init successful?")
 
 			resizeErr := waitForPoolToBeResized(expectedSize, rebootPoolID, isjournal)
 			dash.VerifyFatal(resizeErr, nil, fmt.Sprintf("Verify pool [%s] on node [%s] expansion using auto", rebootPoolID, restartDriver.Name))
+			poolExpandLock.Unlock()
+			wg.Wait()
+
+			done <- true
+			close(done)
```

Comment:
this code is not guaranteed to be executed, and because of that, poolStatusChecker might run indefinitely